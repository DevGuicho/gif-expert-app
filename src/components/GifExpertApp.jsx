import React, { useState } from 'react';
import AddCategory from './AddCategory/index';
import GifGrid from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState(defaultCategories);
  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory categories={categories} setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
