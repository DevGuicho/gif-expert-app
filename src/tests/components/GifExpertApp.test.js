import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../../components/GifExpertApp';

describe('Pruebas en <GifExpertApp />', () => {
  test('deber renderizar componente correctamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
    const categorias = ['One Punch', 'Dragon Ball Z'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categorias} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categorias.length);
  });
});
