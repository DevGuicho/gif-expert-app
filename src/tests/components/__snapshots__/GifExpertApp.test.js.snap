// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    categories={
      Array [
        "One Punch",
        "Dragon Ball Z",
      ]
    }
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> deber renderizar componente correctamente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    categories={Array []}
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
